protocol/src/lib.rs

```diff
@@ -111,25 +113,14 @@ impl fmt::Display for Error {
                 write!(f, "More than 4095 bytes of garbage in the handshake.")
             }
             Error::HandshakeOutOfOrder => write!(f, "Handshake flow out of sequence."),
-            Error::SecretGeneration(e) => write!(f, "Cannot generate secrets: {}.", e),
-            Error::Decryption(e) => write!(f, "Decrytion error: {}.", e),
+            Error::SecretGeneration(e) => write!(f, "Cannot generate secrets: {:?}.", e),
+            Error::Decryption(e) => write!(f, "Decrytion error: {:?}.", e),
         }
     }
 }
 
 #[cfg(feature = "std")]
-impl std::error::Error for Error {
```

Comment:
Why did this get nerfed?

Reply:
Ah, I'll mention it more in the PR description, but had a lint tell me we were essentially re-writing the default impl.